### PR TITLE
chore(lint): move eslint-plugin-fxa into the monorepo

### PR DESCRIPTION
**Pull request by chenba**
Because:
 - we don't want to maintain the fxa eslint plugin in another repo

This commit:
 - move the plugin into the monorepo

### CODE REVIEW

**Comment by chenba** [comment]
Moved eslint-plugin-fxa into packages.

File: packages/fxa-auth-server/package.json [hunk 1]
```diff
@@ -179,7 +179,6 @@
     "esbuild": "^0.14.2",
     "esbuild-register": "^3.2.0",
     "eslint": "^7.32.0",
-    "eslint-plugin-fxa": "^2.0.2",
     "fxa-shared": "workspace:*",
```
Comment (pdehaan) [inline packages/fxa-auth-server/package.json:182]: Unrelated and barely interesting, but this is duplicated on L96 above as a dependency as well:

https://github.com/mozilla/fxa/blob/3e749085c9207a7a1724a3dcb15c0a774838fe14/packages/fxa-auth-server/package.json#L96

(I was wondering why my `git grep "workspace:" packages/*/package.json | sort | uniq | wc -l` returned off-by-one results depending on whether I used `uniq` or not.)

**Review by pdehaan — APPROVED** [review]
Seems to work for me locally (but then again it did last time and only started failing in CI/Docker/multiverse). 🤷 

Tales From a Fresh Clone:
```sh
pwd # /tmp/fxa
yarn workspaces foreach -v run lint # WORKS (lints everything from root dir)
yarn workspace fxa-settings lint # WORKS (lints single package from root dir)

cd packages/fxa-settings
yarn lint # WORKS (lints single package from within package dir)
npx eslint src/pages/ResetPassword/index.tsx # WORKS (lints a single file)

cd packages/fxa-react
yarn lint # WORKS (but also uses `"extends": ["plugin:fxa/recommended"]`)
```

&hellip; where the full list of packages that use custom eslint-plugin-fxa configs is:
```sh
git grep -n "plugin:fxa/" packages | cat

packages/123done/.eslintrc.json:2:  "extends": ["plugin:fxa/server"],
packages/browserid-verifier/.eslintrc.json:2:  "extends": ["plugin:fxa/recommended"],
packages/eslint-plugin-fxa/lib/configs/client.js:8:  extends: 'plugin:fxa/recommended',
packages/eslint-plugin-fxa/lib/configs/recommended.js:6:  extends: ['plugin:fxa/default', 'prettier'],
packages/eslint-plugin-fxa/lib/configs/server.js:8:  extends: 'plugin:fxa/recommended',
packages/fortress/.eslintrc.json:2:  "extends": ["plugin:fxa/server"],
packages/fxa-auth-server/.eslintrc.json:20:  "extends": ["plugin:@typescript-eslint/recommended", "plugin:fxa/server", "prettier"],
packages/fxa-content-server/.eslintrc:2:  "extends": ["plugin:fxa/client"],
packages/fxa-customs-server/.eslintrc:2:  "extends": ["plugin:fxa/client"],
packages/fxa-geodb/.eslintrc:2:  "extends": ["plugin:fxa/server"],
packages/fxa-react/.eslintrc.json:2:  "extends": ["plugin:fxa/recommended"],
packages/fxa-shared/.eslintrc:2:  "extends": ["plugin:fxa/recommended"],
```

File: packages/123done/package.json [hunk 2]
```diff
@@ -33,7 +33,6 @@
   "devDependencies": {
     "audit-filter": "0.5.0",
     "eslint": "^7.32.0",
-    "eslint-plugin-fxa": "^2.0.2",
```
[the remark this answers is not in the document]
Comment (chenba) [reply packages/123done/package.json:36]: I think it's okay to have it as a global dev dependency.  There are other eslint plugins in the root package.json as well.

File: packages/fxa-auth-server/package.json (hunk 1 above)
Comment (dschom) [reply packages/fxa-auth-server/package.json:182]: Just a side note, I think in general we could revisit our dev vs non-dev dependencies... not sure if there is a way to automate this.

File: packages/eslint-plugin-fxa/package.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "eslint-plugin-fxa",
+  "version": "2.0.2",
+  "description": "eslint plugin for Firefox Accounts",
+  "main": "lib/index.js",
+  "scripts": {
+    "test": "mocha"
+  },
+  "dependencies": {
+    "eslint": "6.6.0",
```
Comment (dschom) [inline packages/eslint-plugin-fxa/package.json:10]: I've noticed that our eslint versions are all over the place. Should we consolidate one eslint versions? This is probably a follow up task...

**Review by dschom — COMMENTED** [review]
Looks good to me!

File: packages/eslint-plugin-fxa/package.json (hunk 3 above)
Comment (dschom) [reply packages/eslint-plugin-fxa/package.json:10]: I like what you did with moving the the fxa eslint plugin to the root package.json. Maybe we do the same thing?

**Review by dschom — APPROVED** [review]

File: packages/eslint-plugin-fxa/package.json (hunk 3 above)
Comment (chenba) [reply packages/eslint-plugin-fxa/package.json:10]: Sure, I can file an issue for this.

Comment (pdehaan) [reply packages/eslint-plugin-fxa/package.json:10]: Obligatory <kbd>grep</kbd>:

```sh
git grep '"eslint":' packages/**/package.json | cat

packages/123done/package.json:    "eslint": "^7.32.0",
packages/browserid-verifier/package.json:    "eslint": "^7.32.0",
packages/eslint-plugin-fxa/package.json:    "eslint": "6.6.0",
packages/eslint-plugin-fxa/package.json:    "eslint": ">=6.6.0"
packages/fortress/package.json:    "eslint": "^7.32.0",
packages/fxa-admin-panel/package.json:    "eslint": "^7.32.0",
packages/fxa-admin-server/package.json:    "eslint": "^8.18.0",
packages/fxa-auth-server/package.json:    "eslint": "^7.32.0",
packages/fxa-content-server/package.json:    "eslint": "^7.32.0",
packages/fxa-customs-server/package.json:    "eslint": "^7.32.0",
packages/fxa-event-broker/package.json:    "eslint": "^7.32.0",
packages/fxa-geodb/package.json:    "eslint": "^8.19.0",
packages/fxa-graphql-api/package.json:    "eslint": "^7.32.0",
packages/fxa-payments-server/package.json:    "eslint": "^7.32.0",
packages/fxa-profile-server/package.json:    "eslint": "^8.18.0",
packages/fxa-react/package.json:    "eslint": "^7.32.0",
packages/fxa-settings/package.json:    "eslint": "^7.32.0",
packages/fxa-shared/package.json:    "eslint": "^7.32.0",
```

Or if you prefer uniques:
```sh
git grep -h '"eslint":' packages/**/package.json | sort | uniq

    "eslint": "6.6.0",
    "eslint": ">=6.6.0"
    "eslint": "^7.32.0",
    "eslint": "^8.18.0",
    "eslint": "^8.19.0",
```

**Pull request merged**